File: KataMedication/Ruby/patient.rb
class Patient

  attr_reader :medications

  def initialize()
    @medications = []
  end

  def possession(drugs, days)
      return 0 if drugs.empty?
      day_count = Date.today - days
      drugs.map { |drug| drug.dates_prescribed_in_effective_range(day_count) }.flatten.uniq.size
    end

end

File: KataMedication/Python/patient.py

class Patient:
    
    def __init__(self, medications = []):
        self.medications = medications
    
    def add_medication(self, medication):
        self.medications.append(medication)
    
    def possession(self, medications, days):
        if not medications:
            return 0
        dates = set()
        for medication in medications:
            dates = dates.union(set(medication.dates_prescribed_in_effective_range(days)))
        return len(dates)
